queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/correction.py

```diff
@@ -84,6 +85,16 @@ def correct_business_data(business: Business, correction_filing_rec: Filing,  #
         if business_start_date:
             business.start_date = datetime.datetime.fromisoformat(business_start_date) + timedelta(hours=8)
 
+    # update share structure and resolutions, if any
+    with suppress(IndexError, KeyError, TypeError):
+        share_structure = dpath.util.get(correction_filing, '/correction/shareStructure')
+        shares.update_share_structure(business, share_structure)
```

Comment:
Sorry Lekshmi. I only found `update_share_structure` in the entity-filer project. Is there any other method I wasn't aware of?

Reply:
That is a new method that needs to be written by the person working on this ticket :)

Reply:
I was implementing this new method that compares the differences, and I was a bit confused. Did you mean that we want to append only the differences to `business.resolutions` and not everything from `resolutionDates`? If `resolutionDates` is the same as `business.resolutions`, then the output will be an empty array. I'm not sure whether this will have any impact on other places.

Reply:
For corrections we have to do a compare operation for the objects. If an object in the json has an id that means the object already exists and you need to update it. If there is no id you have to create it. If an object in db with a specific id is not present in the input json that has to be deleted from db. This can be used as a reference
https://github.com/bcgov/lear/blob/main/queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/correction.py#L88

Reply:
Sorry Lekshmi. I am confused how this is possible to update/create/delete the resolution dates because `shareStructure.resolutionDates` is an array of strings, which don't have an associated `id`. For `shareStructure.shareClasses`, the current implementation is working properly, because the frontend doesn't update this object. For `shareStructure.resolutionDates`, the current implementation treats it as an array of strings and append to `business.resolutions` using the model class `Resolution`. 

![image](https://user-images.githubusercontent.com/51458187/190227975-e0aa7944-09d7-4ef4-8498-e6cfc1b6bbba.png)

Reply:
For resolution dates you can delete the ones that are present in db but not in the json and create the ones in json but not in db.

Reply:
Okay. Thanks. I will implement the method in this way. In order to do this, I use `business.resolutions` to check with the json. I will assume `business.resolutions` is an array of the class `Resolution`, from which I can call `resolution_date` from each instance.